src/transformers/models/olmo/modeling_olmo.py

```diff
@@ -302,8 +303,7 @@ def forward(self, x, position_ids):
             emb = torch.cat((freqs, freqs), dim=-1)
             cos = emb.cos() * self.attention_scaling
             sin = emb.sin() * self.attention_scaling
-
-        return cos.to(dtype=x.dtype), sin.to(dtype=x.dtype)
+            return cos, sin
```

Comment:
Rope is mega tricky, if you still have issue, I would check the inv_freqs, they are init on device sometimes, which makes a difference